syntax = "proto3";

package firmware_upgrade;

message srec_line{
    string record_type = 1;
    fixed32 byte_count = 2;
    fixed32 address = 3;
    bytes data = 4;
    fixed32 checksum = 5;
}

message firmware_packet{
    fixed32 checksum = 1;
    srec_line line = 2;
}

message status{
    enum status_enum {
        UNUSED = 0;
        READY = 1;
        ACK = 2;
        REJECT = 3;
    }
    status_enum status = 1;
}